PWGCF/MultiparticleCorrelations/Tasks/O2AnalysisTaskThreePartCorr.cxx

```diff
@@ -0,0 +1,299 @@
+
+          if (CandMass >= 1.10 && CandMass <= 1.13) {
+            if (TrackPID(associate).first == track::PID::Pion) {
+              MECorrRegistry.fill(HIST("hMixLambdaPion_SGNL"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+            } else if (TrackPID(associate).first == track::PID::Kaon) {
+              MECorrRegistry.fill(HIST("hMixLambdaKaon_SGNL"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+            } else if (TrackPID(associate).first == track::PID::Proton) {
+              MECorrRegistry.fill(HIST("hMixLambdaProton_SGNL"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+            }
+          } else {
+            if (TrackPID(associate).first == track::PID::Pion) {
+              MECorrRegistry.fill(HIST("hMixLambdaPion_SB"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+            } else if (TrackPID(associate).first == track::PID::Kaon) {
+              MECorrRegistry.fill(HIST("hMixLambdaKaon_SB"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+            } else if (TrackPID(associate).first == track::PID::Proton) {
+              MECorrRegistry.fill(HIST("hMixLambdaProton_SB"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), trigger.pt(), associate.pt(), TSign, associate.sign());
+            }
+          }
+        }
+      }
+    }
+    // End of the Mixed-events Correlations
+  }
+  PROCESS_SWITCH(ThreePartCorr, processMixed, "Process mixed-event correlations", true);
+
+  //================================================================================================================================================================================================================
+
+  Double_t PhiCorrection(Double_t TriggerPhi, Double_t AssociatePhi)
+  {
+    Double_t dPhi = TriggerPhi - AssociatePhi;
+
+    if (dPhi < (-1. / 2) * M_PI) {
+      dPhi = dPhi + 2 * M_PI;
+    } else if (dPhi > (3. / 2) * M_PI) {
+      dPhi = dPhi - 2 * M_PI;
+    }
+    return dPhi;
+  }
+
+  template <class TrackCand>
+  std::pair<track::PID::ID, Double_t> TrackPID(const TrackCand& Track)
+  {
+    Double_t NSigma[3];
+    NSigma[0] = TMath::Abs(Track.tpcNSigmaPi());
+    NSigma[1] = TMath::Abs(Track.tpcNSigmaKa());
+    NSigma[2] = TMath::Abs(Track.tpcNSigmaPr());
+
+    if (NSigma[0] < std::min(NSigma[1], NSigma[2])) {
+      return std::make_pair(track::PID::Pion, NSigma[0]);
+    } else if (NSigma[1] < std::min(NSigma[0], NSigma[2])) {
+      return std::make_pair(track::PID::Kaon, NSigma[1]);
+    } else if (NSigma[2] < std::min(NSigma[0], NSigma[1])) {
+      return std::make_pair(track::PID::Proton, NSigma[2]);
+    }
+    return std::make_pair(track::PID::Electron, 10);
+  }
```

Comment:
Creating `std::pair` is usually time consuming
Consider if this structure is really needed here
If there is excessive execution time this is a candidate to address